wire-library/wire-runtime-swift/src/main/swift/JSONString.swift

```diff
@@ -26,10 +26,18 @@ public struct JSONString<T : Hashable> : Hashable, Codable {
         switch T.self {
         case is Int64.Type, is Optional<Int64>.Type:
             let container = try decoder.singleValueContainer()
-            self.wrappedValue = Int64(try container.decode(String.self))! as! T
+            if let value = try? container.decode(String.self) {
```

Comment:
We're possibly swallowing real errors here now. Like if the value were actually (unexpectedly) an array, we would set the wrapped value to `.none` and carry on when we should throw the error. A few options I see:
1. Try explicitly decoding an optional if the type is optional, like `try container.decode(Optional<Int64>.self)`. Not totally sure if this works, but if it does that's ideal.
2. Figure out what error gets thrown for `null` and explicitly handle that error as a `.none` optional
3. Try `decodeNil` first, then if that fails use the old logic. Since `decodeNil` will fail in fewer ways we can be relatively sure it's failing because the value is not-null

Note that for a few of these it may make sense to split the optional and non-optional case statements out rather than grouping them like we were doing previously

Reply:
Option 1 does not work, and fails with the following error: `"Expected to decode Int64 but found a string/data instead.", underlyingError: nil))"` because the Ints are in actuality Strings. Opted for option 3 with extension functions on Decoder to make it look like option 1 at the call site.